docs/recipes/hydrating-state.md

```diff
@@ -0,0 +1,100 @@
+# Hydrating State
+
+Often times your application will need to start in a specific
+state. For example: server rendered pages that need to be hydrated in
+a specific state, or applications that save data in `localStorage` for
+return visits.
+
+Microcosm ships with several utilities that make this easy.
+
+## Resetting state
+
+The `reset` method wipes away existing state, replacing it with new
+data. Where domain keys are unspecified, values are replaced with the
+result of `getInitialState`:
+
+```javascript
+import Microcosm from 'microcosm'
+
+let repo = new Microcosm()
+
+repo.addDomain('planets', {
+  getInitialState () {
+    return []
+  }
+})
+
+// The initial state is an empty object
+repo.state.planets // []
+
+// Now reset the state with new data
+repo.reset({ planets: [{ name: "Earth" }]})
+
+// The repo now contains planets
+repo.state.planets // [{ name: "Earth" }]
+
+// Reset without a data payload
+repo.reset()
+
+// Since no planets were provided, the repo has reverted to
+// the initial plents state
+repo.state.planets // []
```

Comment:
Ah, this code example more clearly illustrates the "Where domain keys are unspecified" concept. Still might consider some re-wording.